lib/config/customer_io_config.dart

```diff
@@ -0,0 +1,53 @@
+import '../customer_io_enums.dart';
+import '../customer_io_plugin_version.dart' as plugin_info show version;
+import 'in_app_config.dart';
+import 'push_config.dart';
+
+class CustomerIOConfig {
+  final String source = 'Flutter';
+  final String version = plugin_info.version;
+
+  final String cdpApiKey;
+  final String? migrationSiteId;
+  final Region? region;
+  final CioLogLevel? logLevel;
+  final bool? autoTrackDeviceAttributes;
+  final String? apiHost;
+  final String? cdnHost;
+  final int? flushAt;
+  final int? flushInterval;
+  final InAppConfig? inAppConfig;
+  final PushConfig pushConfig;
+
+  CustomerIOConfig({
+    required this.cdpApiKey,
+    this.migrationSiteId,
+    this.region,
+    this.logLevel,
+    this.autoTrackDeviceAttributes,
+    this.apiHost,
+    this.cdnHost,
+    this.flushAt,
+    this.flushInterval,
+    this.inAppConfig,
```

Comment:
thinking our load, if a user has provided `migrationSiteId` and not `InAppConfig` should we automatically handle it for them?

Reply:
I initially thought the same, but doing it that way would mean losing flexibility in two scenarios:

- Disabling in-app messaging wouldn't be possible if `migrationSiteId` is provided
- If a user creates a new `siteId`, it could cause issues since `migrationSiteId` is supposed to remain the same, while in-app feature would require the new `siteId`

Keeping them separate gives us better control in these cases.